src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```
@@ -310,6 +323,40 @@
     <MakeDir Directories="$(ProjectSpecificConfDir)" />
   </Target>
 
+  <!-- **************************************************************************** -->
+  <!-- In case of Razor compilation the protobuf files generated during the core compilation step are already moved into a temporary folder.
+       This target moves the protobuf files generated during the razor core compile step into a .razor directory, and alsomoves the protobuf files
+       from the temporary directory back to their original place. This target also takes care about generating the ProjectInfo.xml file for the razor part of the compilation.
+  -->
+  <!-- **************************************************************************** -->
+  <Target Name="CopyProjectInfoFileInPresenceOfRazor" AfterTargets="WriteSonarQubeProjectData" Condition=" $(TemporaryProjectSpecificOutDir) != '' ">
+    <PropertyGroup>
+      <RazorProjectSpecificOutDir>$(ProjectSpecificOutDir).razor</RazorProjectSpecificOutDir>
```

`Sonar` prefix as it's a new property. And all of them should have `Sonar` prefix one day.
I prefer `.Razor` over `.razor` for beatury reasons.
```suggestion
      <SonarRazorProjectSpecificOutDir>$(ProjectSpecificOutDir).Razor</SonarRazorProjectSpecificOutDir>
```